.github/workflows/release.yml

```diff
@@ -39,19 +26,52 @@ jobs:
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials registry | PRIME_REGISTRY ;
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials username | PRIME_REGISTRY_USERNAME ;
           secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials password | PRIME_REGISTRY_PASSWORD
-    - name: Build and push all image variations
+    - name: Check SemVer Characteristics
+      id: semver_check
+      run: |
+        tag="${{ github.ref_name }}"
+        if [[ "$tag" == v* ]]; then
+          tag="${tag:1}"
+        fi
+        has_prerelease() { [[ "$1" =~ ^[0-9]+\.[0-9]+\.[0-9]+- ]]; }
+        has_build_meta() { [[ "$1" =~ \+ ]]; }
+
+        if has_prerelease "$tag"; then
+          echo "HAS_PRERELEASE=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "HAS_PRERELEASE=false" >> "$GITHUB_OUTPUT"
+        fi
+
+        if has_build_meta "$tag"; then
+          echo "HAS_BUILD_META=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "HAS_BUILD_META=false" >> "$GITHUB_OUTPUT"
+        fi
+    - name: Build and push image (DockerHub and PrimeStaging)
       uses: rancher/ecm-distro-tools/actions/publish-image@master
       with:
         image: shell
-        tag: ${{ github.ref_name }}${{ matrix.tag-suffix }}
-        platforms: ${{ matrix.platforms }}
+        tag: ${{ github.ref_name }}
+        platforms: "linux/amd64,linux/arm64"
 
         public-registry: docker.io
         public-repo: rancher
         public-username: ${{ env.DOCKER_USERNAME }}
         public-password: ${{ env.DOCKER_PASSWORD }}
 
-        prime-registry: ${{ env.PRIME_REGISTRY }}
+        prime-registry: '<INSERT PRIME STAGING REGISTRY HERE>'
         prime-repo: rancher
         prime-username: ${{ env.PRIME_REGISTRY_USERNAME }}
         prime-password: ${{ env.PRIME_REGISTRY_PASSWORD }}
+    - name: Build and push image (PrimeProd)
```

Comment:
@pjbgf - unless I'm overlooking something that would cause big issues, I think this general idea (check tag SemVer and conditionally push to Stage, or Stage+Prod) could be done at the EIO action level. And essentially as a user if `push-to-prime` is set true, it would handle both of these.

The only major concern I have for that is how Stage credentials are handled - worst case I imagine repos needing new secrets to vault and an addition of ` prime-stage-username` and ` prime-stage-password` to cover that?

Reply:
Not sure we need two separate creds, as both are "production" in the sense that they serve customers. Potentially SCC can expand the access to the additional repository.

Reply:
> Potentially SCC can expand the access to the additional repository.

Agreed, that sounds ideal.